test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -849,6 +849,58 @@ def test_avgpool_3d_ceil(self):
         x = torch.randn(20, 16, 50, 44, 31)
         self.run_test(model, x)
 
+    def test_floating_point(self):
+        class FloatingPoint(torch.jit.ScriptModule):
+            @torch.jit.script_method
+            def forward(self, x):
+                if x.is_floating_point():
+                    return x.new_zeros(x.shape)
+                return x.new_zeros(x.shape)
+
+        x = torch.randn(2, 3, 4)
+        self.run_test(FloatingPoint(), x)
+
+        class FloatingPoint(torch.jit.ScriptModule):
+            @torch.jit.script_method
+            def forward(self, x):
+                if x.size(0) > 1:
```

Comment:
why need this condition in the test case, is it affecting the the test case somehow?

Reply:
I needed to create a case where a type could not be inferred from torch traced graph and nested if statement represent one of those cases.